### ISSUE TITLE
std.json.stringify does not support std.hash_map

### ISSUE THREAD

**Issue by Jarred-Sumner** [ziglang/zig#8660]
The specific error is this:
```
Unable to stringify type '[*]std.hash_map.Metadata'
```
![image](https://user-images.githubusercontent.com/709451/116766331-c5b8f980-a9de-11eb-8a96-dbd8c0063f4d.png)

I believe the fix would be to implement 
```zig
pub fn jsonStringify(options: StringifyOptions, stream: WriteStream)
```

It should output a JSON object (rather than something like an array of keys and an array of values).

**Comment by daurnimator** [comment]
> `Unable to stringify type '[*]std.hash_map.Metadata'`

`[*]T` is "unknown number of": there is no way for zig to serialise that to json.

---------------

However, as a general task, implementing a `jsonStringify` on `StringHashMap` sounds useful; though it brings up questions of robustness in the face of e.g. invalid utf8 keys

**Comment by r00ster91** [comment]
Can we please also add support for `StringHashMap` or similar in the `type` to `std.json.parse` (the opposite)? Sometimes you don't know what the key will be in the object. Should I open an issue for this?

**Comment by nektro** [comment]
for parsing unknown keys use `std.json.Parser`

**Comment by thejoshwolfe** [comment]
We've got a dedicated `std.json.ArrayHashMap` from https://github.com/ziglang/zig/pull/16366 available in version Zig 0.11.0. Please reopen if this doesn't solve your usecase.
